webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/services/LogDisplayer.scala

```diff
@@ -172,34 +172,55 @@ class LogDisplayer(
     """) // JsRaw ok, escaped
   }
 
+  def refreshData(nodeId: NodeId, reports: => Seq[Reports], tableId: String): JsCmd = {
+    def getDirectiveName(directiveId: DirectiveId): Box[String] = {
+      configRepository
+        .getDirective(directiveId)
+        .map(_.map(_.directive.name).getOrElse(directiveId.serialize))
+        .toBox
+    }
+
+    def getRuleName(ruleId: RuleId): Box[String] = {
+      ruleRepository.get(ruleId).map(x => x.name).toBox
+    }
+
+    val data = LogDisplayer.getReportsLineForNode(reports, getDirectiveName, getRuleName)
+
+    OnLoad(JsRaw(s"""refreshTable("${StringEscapeUtils.escapeEcmaScript(tableId)}",${data});"""))
```

Comment:
`${data}` will represent the `data` as string but it is a case class, however we want it's json representation with `data.json.toJsCmd` !